R/ParameterIdentification.R

```diff
@@ -482,28 +482,170 @@ ParameterIdentification <- R6::R6Class(
         x$maxValue
       }), use.names = FALSE)
 
-      # The method and method-specific control options can be passed
-      # to the `PIConfiguration` object. The `FME::modFit()` function allows to
-      # use the following algorithms:
-      # * `bobyqa` is a gradient-free quadratic approximation method
-      # * * `rhobeg` is the initial value of the trust region radius. Set to one tenth of the greatest expected variable change
-      # * * `maxfun` is the number of function evaluations
-      # * `Marq` is the Levenberg-Marquardt method
-      # * * `ftol`, `ptol` and `gtol` control the termination criteria
-      # * * `maxfev` is the number of function evaluations
-      # * * `maxiter` is the number of iterations
-      results <- FME::modFit(f = private$.targetFunction, p = startValues, lower = lower, upper = upper, method = private$.configuration$algorithm, control = private$.configuration$algorithmOptions)
-
-      # Calculation of confidence intervals
-      # Sigma values are standard deviations of the estimated parameters. They are
-      # extracted from the estimated hessian matrix through the summary function.
-      # The 95% confidence intervals are defined by two sigma values away from the
-      # point estimate. The coefficient of variation (CV) is the ratio of standard
-      # deviation to the point estimate.
-      sigma <- as.numeric(summary(results)[["par"]][, "Std. Error"])
-      results$lwr <- results$par - 1.96 * sigma
-      results$upr <- results$par + 1.96 * sigma
-      results$cv <- sigma / abs(results$par) * 100
+      # Depending on the method argument in the `PIConfiguration` object, the
+      # actual optimization call will use one of the underlying optimization routines
+      message(paste0("Running optimization algorithm: ", private$.configuration$algorithm))
+      results <- NULL
+      if (private$.configuration$algorithm %in% c("bobyqa", "Marq")) {
+        time <- system.time(results <- FME::modFit(f = private$.targetFunction, p = startValues, lower = lower, upper = upper, method = private$.configuration$algorithm, control = private$.configuration$algorithmOptions))
+
+        # Sigma values are standard deviations of the estimated parameters. They are
+        # extracted from the estimated hessian matrix through the summary function.
+        # The 95% confidence intervals are defined by two sigma values away from the
+        # point estimate. The coefficient of variation (CV) is the ratio of standard
+        # deviation to the point estimate.
+        sigma <- as.numeric(summary(results)[["par"]][, "Std. Error"])
+        results$lwr <- results$par - 1.96 * sigma
+        results$upr <- results$par + 1.96 * sigma
+        results$cv <- sigma / abs(results$par) * 100
+
+        results$elapsed <- time[[3]]
+      }
+      if (private$.configuration$algorithm %in% c("Nelder-Mead", "BFGS", "CG", "L-BFGS-B", "SANN")) {
```

Comment:
Why is this a separate `if`, i.e., why not use the call to `FME::modFit()`?

Reply:
This allows to test the differences between calling Nelder-Mead from `{FME}` vs calling Nelder-Mead from `{optim}`.